packages/compass-collection/src/stores/index.ts

```diff
@@ -157,6 +156,7 @@ store.onActivated = (appRegistry: AppRegistry) => {
    */
   appRegistry.on('instance-destroyed', () => {
     store[kInstance] = null;
+    store.dispatch(reset());
```

Comment:
Do I understand it right that this would be enough to make the guard on line 129 (`if (!store[kInstance]) {`) to pass and execute the code inside the `if`?

I'm not super sure how these events are tight together but shouldn't we also clean up the listeners that we attached on `instance.on` and `instance.dataLake.on` here?

Reply:
They are auto cleaned up when instance is destroyed on disconnect so no need to do this manually